boot/zephyr/main.c

```diff
@@ -202,9 +203,11 @@ void main(void)
 #endif
 
 #ifdef CONFIG_BOOT_WAIT_FOR_USB_DFU
-    BOOT_LOG_INF("Waiting for USB DFU");
-    wait_for_usb_dfu();
-    BOOT_LOG_INF("USB DFU wait time elapsed");
+    if (is_entering_usb_dfu()) {
```

Comment:
need default implementation of is_entering_usb_dfu() for any board, or #ifdef'ry

Reply:
How about like this?

* common.c
```
__attribute__((weak)) int is_entering_usb_dfu(void)
{
        /* Nothing to do */
        return 1;
}
```

This definition is built by default.